File: v6-backup/fix_final.py
path = r'C:\Users\taidu\.openclaw\workspace\projects\trendy-cafe-game\index.html'

with open(path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Fix line 741 (0-indexed: 740)
# The issue is the onclick="buyRecipe('" + item.id + "')"> which has conflicting quotes
# Just replace it with onclick=buyRecipe(item.id)
for i, line in enumerate(lines):
    if 'buyRecipe(' in line and 'Unlock' in line:
        lines[i] = line.replace(
            'onclick="buyRecipe(\'" + item.id + "\')">',
            'onclick=buyRecipe(item.id)>'
        )
        print('Fixed line ' + str(i+1))
        break

with open(path, 'w', encoding='utf-8') as f:
    f.writelines(lines)

print("Done!")

File: fix_final.js
const fs = require('fs');
let html = fs.readFileSync('C:/Users/taidu/.openclaw/workspace/projects/trendy-cafe-game/index.html', 'utf8');

// The problem: line 207 ends with `}}` (two closing braces)
// it should end with `}` (one closing brace)
// Pattern: `setTimeout(startGame,0);}}` followed by newline
// Fix: change to `setTimeout(startGame,0);}`

html = html.replace(
  /(setTimeout\(startGame,0\);)\}\}\s*\n<\/script>/,
  '$1}\n'
);

// Verify
const scriptMatch = html.match(/<script>([\s\S]*?)<\/script>/);
const script = scriptMatch[1];
let bl = 0;
for (const ch of script) { if (ch === '{') bl++; if (ch === '}') bl--; }
console.log('Balance:', bl);
console.log('JS syntax valid:', (() => { try { new Function(script); return true; } catch(e) { return false; } })());

fs.writeFileSync('C:/Users/taidu/.openclaw/workspace/projects/trendy-cafe-game/index.html', html);
console.log('Fixed!');
